### PR TITLE
Upgrade RESTEasy to 4.2.0.Final

### PR DESCRIPTION
**Pull request by asoldano**

### CODE REVIEW

**Comment by geoand** [comment]
Relaunched tests to see if the failures were just temporary

**Comment by geoand** [comment]
Seems like there is a dependency version issue in the TCKs:

```
[WARNING] 
Dependency convergence error for org.jboss.arquillian.test:arquillian-test-spi:1.4.1.Final paths to dependency are:
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-io.quarkus:quarkus-arquillian:999-SNAPSHOT
    +-org.jboss.arquillian.container:arquillian-container-test-spi:1.4.1.Final
      +-org.jboss.arquillian.test:arquillian-test-spi:1.4.1.Final
and
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-io.quarkus:quarkus-arquillian:999-SNAPSHOT
    +-org.jboss.arquillian.junit:arquillian-junit-container:1.4.1.Final
      +-org.jboss.arquillian.junit:arquillian-junit-core:1.4.1.Final
        +-org.jboss.arquillian.test:arquillian-test-spi:1.4.1.Final
and
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-io.quarkus:quarkus-arquillian:999-SNAPSHOT
    +-org.jboss.arquillian.junit:arquillian-junit-container:1.4.1.Final
      +-org.jboss.arquillian.test:arquillian-test-spi:1.4.1.Final
and
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-io.quarkus:quarkus-arquillian:999-SNAPSHOT
    +-org.jboss.arquillian.junit:arquillian-junit-container:1.4.1.Final
      +-org.jboss.arquillian.test:arquillian-test-impl-base:1.4.1.Final
        +-org.jboss.arquillian.test:arquillian-test-spi:1.4.1.Final
and
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-org.eclipse.microprofile.config:microprofile-config-tck:1.3
    +-org.jboss.arquillian.testng:arquillian-testng-container:1.1.13.Final
      +-org.jboss.arquillian.testng:arquillian-testng-core:1.1.13.Final
        +-org.jboss.arquillian.test:arquillian-test-spi:1.1.13.Final
and
+-io.quarkus:quarkus-tck-microprofile-config:999-SNAPSHOT
  +-org.eclipse.microprofile.config:microprofile-config-tck:1.3
    +-org.jboss.arquillian.testng:arquillian-testng-container:1.1.13.Final
      +-org.jboss.arquillian.test:arquillian-test-spi:1.1.13.Final
```

which results in:

```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
```

So I guess we need to enforce the Arquillian version in the BOM?

**Review by geoand — APPROVED** [review]
:+1: 

I would however like to have another set of eyes on this, @kenfinnigan  or @mkouba  perhaps?

File: tcks/pom.xml
Comment (kenfinnigan) [inline tcks/pom.xml]: Would it be better if we excluded any arquillian dependencies from the TCK dependencies instead of doing it this way?

File: bom/runtime/pom.xml
Comment (kenfinnigan) [inline bom/runtime/pom.xml]: What changed in RESTEasy to need EL and Commons IO dependencies?

File: bom/runtime/pom.xml
Comment (kenfinnigan) [inline bom/runtime/pom.xml]: Any reason the API needs to be a direct dependency?

File: tcks/pom.xml
Comment (asoldano) [reply tcks/pom.xml]: I haven't tried that to be honest, but when looking for build failure details in the log I think I noticed that the TCKs here were using more of Arquillian than what was clashing with the rest of the project, so completely excluding arquillian might not work and I thought relying on depenencyManagement would have been simpler / less fragile.

File: bom/runtime/pom.xml
Comment (asoldano) [reply bom/runtime/pom.xml]: This is doing dependency management. The actual dependency is already set elsewhere in Quarkus, for example in the quarkus-arc extension. The problem is that previously it was relying on resteasy to control that dependency version.

**Comment by kenfinnigan** [comment]
I just found out that the MP TCKs aren't actually being run at present, so will need to put this on hold until that's fixed

**Comment by geoand** [comment]
Unless I am missing something, the TCKs are being run (the build is [here](https://dev.azure.com/quarkus-ci/quarkus/_build/results?buildId=5290)).

I see the following:

```
2019-08-04 21:57:16,470 INFO  [io.qua.arc.pro.BeanProcessor] (build-4) Found unrecommended usage of private members (use package-private instead) in application beans:
	- @Inject field org.eclipse.microprofile.config.tck.EmptyValuesTest#config,
	- @Inject field org.eclipse.microprofile.config.tck.EmptyValuesTest#emptyValuesBean
2019-08-04 21:57:16,503 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 111ms
2019-08-04 21:57:16,517 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 0.013s. 
2019-08-04 21:57:16,517 INFO  [io.quarkus] (main) Installed features: [cdi]
2019-08-04 21:57:16,524 INFO  [io.quarkus] (main) Quarkus stopped in 0.000s
[INFO] Tests run: 70, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 11.242 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 70, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ quarkus-tck-microprofile-config ---
```

```
2019-08-04 21:57:31,071 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-08-04 21:57:31,282 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 211ms
2019-08-04 21:57:31,304 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 0.020s. Listening on: http://[::]:8081
2019-08-04 21:57:31,304 INFO  [io.quarkus] (main) Installed features: [cdi, smallrye-health]
{"status":"UP","checks":[{"name":"successful-check","status":"UP"}]}
{"status":"UP","checks":[{"name":"successful-check","status":"UP"}]}
2019-08-04 21:57:31,326 INFO  [io.quarkus] (main) Quarkus stopped in 0.004s
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 11.208 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ quarkus-tck-microprofile-health ---
```

```
2019-08-04 21:58:03,884 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-08-04 21:58:04,037 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 153ms
2019-08-04 21:58:04,063 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 0.024s. 
2019-08-04 21:58:04,063 INFO  [io.quarkus] (main) Installed features: [cdi, rest-client]
2019-08-04 21:58:04,080 INFO  [io.quarkus] (main) Quarkus stopped in 0.000s
[INFO] Tests run: 53, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.709 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 53, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:3.1.0:jar (default-jar) @ quarkus-tck-microprofile-rest-client ---
```

**Comment by asoldano** [comment]
> I just found out that the MP TCKs aren't actually being run at present, so will need to put this on hold until that's fixed

There seem to be dedicated CI jobs for those.

**Comment by kenfinnigan** [comment]
Yeah, I misinterpreted things, they are running fine

File: bom/runtime/pom.xml
Comment (kenfinnigan) [reply bom/runtime/pom.xml]: It was relying on RESTEasy? I find that odd considering it would be brought in through SmallRye Context Propagation

**Comment by geoand** [comment]
> Yeah, I misinterpreted things, they are running fine

Yeah, it's easy to get confused the way Azure shows the logs

File: tcks/pom.xml
Comment (kenfinnigan) [reply tcks/pom.xml]: Ok, but the version of Arquillian is also present in the test framework module so it would be good if we defined a property somewhere for it

Comment (asoldano) [reply tcks/pom.xml]: OK, just added in build-parent/pom.xml which is what the tcks and test-framework have in common.

File: bom/runtime/pom.xml
Comment (asoldano) [reply bom/runtime/pom.xml]: The odds of Maven, I'd say. Maybe the dependency path through RESTEasy was simply shorter and hence considered first by Maven.

**Comment by geoand** [comment]
@kenfinnigan do the latest changes @asoldano made address your comments?

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**
